### ISSUE TITLE
{ is annoying

### ISSUE THREAD

**Issue by Schabaani** [intel/intel-one-mono#29]
`{` symbol is annoying.

![intel-font](https://github.com/intel/intel-one-mono/assets/16123923/95d7e0be-5a1c-4fcb-aee4-bae095ee09a3)

**Comment by kay54068** [comment]
Agreed, the { symbol is quite peculiar. What is the intent behind its design?

**Comment by Schabaani** [comment]
Sure! However, this is annoying! I am sure they can make it better.

**Comment by rwmitchell** [comment]
What exactly makes it annoying?  The first time I heard someone refer to these as something other than curly-braces, they called them buzzard wings.  That was 30+ years ago.  This symbol still seems to match that.

**Comment by yavorski** [comment]
It is actually quite cool, I like it.
Thin `@` is more annoying imho.

**Comment by Schabaani** [comment]
@rwmitchell and @yavorski 

Compare to other symbols and other fonts, it has sharp edges that make it an angry symbol. Also, it is not smiling anymore :D    `:}`   .

![symbols](https://github.com/intel/intel-one-mono/assets/16123923/d1be90cb-7c23-4193-a1fa-cd20cad06a76)
![smiling](https://github.com/intel/intel-one-mono/assets/16123923/d1dcd8aa-3b42-4935-92f1-d6e55528647a)

**Comment by rwmitchell** [comment]
it is also much more obvious as to what it is.  The one on the second line looks too much like () for me.

**Comment by tkircher** [comment]
The defining characteristic of a curly bracket is that, in the left curly bracket case, from middle point to top, and from middle point to bottom, the stroke is always moving left to right. It is objectively incorrect for it to curl back. So if you want to exaggerate the features of a curly bracket, do not make it into a different glyph entirely.

**Comment by jnfingerle** [comment]
> [...] defining characteristic [...] objectively [...]

@tkircher, you're obviously referring to some commonly accepted definition of curly brackets, that I'm not aware of, maybe a standard by a standard body? Could you provide us with a link to said definition?

**Comment by RGB-es** [comment]
> The defining characteristic of a curly bracket is that, in the left curly bracket case, from middle point to top, and from middle point to bottom, the stroke is always moving left to right. It is objectively incorrect for it to curl back. So if you want to exaggerate the features of a curly bracket, do not make it into a different glyph entirely.

Beside the fact that there is not such a thing as "objective truth" on type design, and accepting that Intel One Mono goes far beyond any other font on that direction, there are other examples of fonts where curly brackets "curl back" by different amounts: even iconic designs such as  the original IBM Courier do that to an extent (small, true, but clearly present, you can see it in the [Wikipedia article](https://en.wikipedia.org/wiki/Courier_(typeface))), but also Fantasque Sans Mono, Luxi Mono, etc. You find it also in proportional fonts such as EBGaramond, ETbb, Gentium (where the middle point even goes a bit downwards for the left bracket and upwards for the right one), Vollkorn, etc. 

Again, for Intel One Mono is far more pronounced than for any other font, but it's not new. Just my 2 cents.

**Comment by gburke-ppb** [comment]
It's designed that way so that people who are visually impaired can more easily see that it is, indeed, a curly bracket, and not something else, based on feedback from visually impaired people.

It should stay as it is, as the font is designed for them.

**Comment by rwmitchell** [comment]
if it was possible to have only one font that made everyone happy, we'd have only one font.  we don't, we have lots of fonts to choose from.

having said that, it does appear to be missing some symbols I normally have in my vim airline status :-)

**Comment by jeffalder** [comment]
> if it was possible to have only one font that made everyone happy, we'd have only one font. we don't, we have lots of fonts to choose from.

This. If you don't like the curly braces, then don't use this font.

https://www.codingfont.com/ can run an elimination bracket between pairs of fonts until it helps narrow down to one you like. This font is not yet included in its list. 

When I ran this, I landed on Hack. I'm testing this font to see if I like it better. (So far, I still like Hack better.)

**Comment by alexmyczko** [comment]
I landed on https://github.com/blobject/agave

**Comment by tkircher** [comment]
> If you don't like the curly braces, then don't use this font.

The better solution is for everyone to share their opinion, discuss it, and if necessary, fork the font to modify it as desired. More readable fonts benefit everyone.

As far as the typography of curly brackets, here's a decent article:

https://macrotypography.blogspot.com/2015/11/curly-braces.html

Considering how many people find those two glyphs alone obnoxious, and love the rest of the font, it should at least be a matter of consideration. It's ironic though, to argue on one hand that typography is purely subjective, while also arguing on the other hand that empirically, some people find other fonts more difficult to read.

**Comment by jnfingerle** [comment]
> Considering how many people find those two glyphs alone obnoxious, and love the rest of the font, it should at least be a matter of consideration.

At the time of my writing the first item in this discussion has 4 👍 , but 6 👎. This should be part of any consideration, as well.

> It's ironic though, to argue on one hand that typography is purely subjective, 

In fact, nobody did that. Saying, correctly,  that there is no objective truth, especially in how glyphs are supposed to look, doesn't rule out that you may gain some objective insights into certain aspects. It's still subjective,  how you decide to weigh those objective insights. In other words: There might be objective data, but no objective truth in what to conclude from that data. The latter is rather part of the design decisions.

**Comment by rwmitchell** [comment]
@jnfingerle do you pick and choose and much?  Just below that it has:

<img width="440" alt="image" src="https://github.com/intel/intel-one-mono/assets/15944841/7b49e844-a194-4768-94bc-21a512ff9617">

That shows 13 in favor of the current glyph and only 1 in favor.  Personally I think that shows how little this ad-hoc voting really means, but there it is - the current glyph is clearly the winner.

**Comment by jnfingerle** [comment]
@rwmitchell Maybe it's me not being a native speaker, but I'm getting the impression that you disagree with something I wrote, but I fail to see where. Could you clarify?

**Comment by rwmitchell** [comment]
<img width="809" alt="image" src="https://github.com/intel/intel-one-mono/assets/15944841/ad4da2e1-d67b-400b-8175-573e18f53130">

Right above.  You chose the up/down thumbs for one post to make your point that the current braces were not well liked, when only a couple posts below it shows the exact opposite.  Being first post doesn't make the first post special, just makes it first.

**Comment by jnfingerle** [comment]
@rwmitchell wrote

> Considering how many people find those two glyphs alone obnoxious

, to which I replied that more people disagreed than agreed with the first post's

> { symbol is annoying.

The first post *is* special in that it's by the original author of this issue, that's why I referred to that post and didn't write a PhD about voting on Github in general.  

Anyway, you misinterpret what I wrote, my point never was and never will be "that the current braces were not well liked". My point was and is "even if some people don't like them, even more people disagree with that".

**Comment by Mempler** [comment]
I would appreciate a alternative to the { aswell. I'm someone who does a lot of C/++, and Rust code, there, you see a lot of `{`. They kinda hurt my eyes, not in a sense of annoyance but rather in General, as if I'm forced to focus on them.

I don't agree that it should be changed in the first place, but create an alternative as an otf variable for people like me. People can then choose which one you would like to use.

**Comment by deepcoder** [comment]
They do kind of make me think I am in the 'bat cave' Robin.... ;-}

**Comment by Mempler** [comment]
Oh for sure! LOL

**Comment by alexmyczko** [comment]
@deepcoder nethack.org?

**Comment by JohnVillalovos** [comment]
The unique style (in my experience) of the curly braces is one of the main reasons I decided to use this font. Please keep them.

**Comment by Schabaani** [comment]
> The unique style (in my experience) of the curly braces is one of the main reasons I decided to use this font. Please keep them.

I agree with you. It is unique. I don't want the symbol to miss its uniqueness. But I believe they can improve it.

**Comment by tkuraku** [comment]
I think the curly braces are great as is. Symbols are very nicely differentiated.

**Comment by sonulohani** [comment]
The curly braces are great in this font, highly readable. It doesn't need any change

**Comment by abhibeckert** [comment]
I also don't like how much it goes back on itself and think it could be improved. I'd love to see a few variants created and proposed to the community, in order to decide on a new glyph that everyone is happy with.

I'm not qualified to create those variants, but as a quick concept - here's how I'm imagining the glyph could look could look rendered at 13px after amending it to be closer to a traditional curly brace:

![image](https://github.com/intel/intel-one-mono/assets/266088/91103da9-df8a-400e-8a6d-b820b48a7cb9)

**Comment by jnfingerle** [comment]
@abhibeckert The font was designed for glyphs to be easily distinguishable, and people use this font for exactly that reason. There are lots of different monospaced fonts around with different design goals, so if one doesn't agree with those goals, one can use a different font. That said, I have to assume that you agree with the design goals of this font, but I fail to see how your proposal helps to better achieve those goals. If anything, it makes things worse.

**Comment by abhibeckert** [comment]
I have to assume that you agree with the design goals of this font, but I fail to see how your proposal helps to better achieve those goals

**Comment by jnfingerle** [comment]
@abhibeckert I didn't propose anything, but nice trolling on your part.

**Comment by abhibeckert** [comment]
(sorry not sure what happened - github submitted my comment while I was half way through writing it - I've deleted the partial comment)

> I have to assume that you agree with the design goals of this font, but I fail to see how your proposal helps to better achieve those goals

I absolutely do, and I'm contributing to this discussion because I'm using this font and love it.

My own vision isn't perfect, especially towards the end of a long day. I have tested my proposal and it is at least with my vision on my display and the font size I use, perfectly distinct. More work would be required to verify it in other cases. I definitely wouldn't want any change to be made that causes problems for some people who use the font.

**Comment by jnfingerle** [comment]
@abhibeckert Ok, I deleted my reply to said partial comment.

Yes, I am in favour of testing, if a change should be considered. But, if you calculate a "distance" between the current { and [ in comparison to your example, this distance has been reduced. That leads me to believe that your example would have worse results in testing. Obviously, it's only guessing at this point.

**Comment by tkircher** [comment]
> @abhibeckert The font was designed for glyphs to be easily distinguishable, and people use this font for exactly that reason. There are lots of different monospaced fonts around with different design goals, so if one doesn't agree with those goals, one can use a different font. That said, I have to assume that you agree with the design goals of this font, but I fail to see how your proposal helps to better achieve those goals. If anything, it makes things worse.

The curly bracket makes this font totally unreadable. It literally gives me a migraine looking at it. I use this font with a different, more correct glyph in place of the poorly designed one, and it's basically perfect now. Nobody's going to force you or Intel to change anything, but you can't pretend that this isn't an issue.

**Comment by tkuraku** [comment]
> The curly bracket makes this font totally unreadable. It literally gives me a migraine looking at it. I use this font with a different, more correct glyph in place of the poorly designed one, and it's basically perfect now. Nobody's going to force you or Intel to change anything, but you can't pretend that this isn't an issue.

I use this font all day everyday. The curly bracket is pleasant for me to look at and use. I'm not going to pretend it isn't an issue for some, but you also can't pretend that it is a universal issue. I like it how it is.

**Comment by jnfingerle** [comment]
>  you can't pretend

@tkircher I'm sorry to hear about your medical problems, and it's great to hear that you solved those. But you implying that I didn't argue in good faith won't help the likelihood of me agreeing with you. So just to state the obvious: The point of so-called "correctness" has already been discussed, and I did not pretend anything.

**Comment by georgalis** [comment]
```

    \|/ ____ \|/
    "@'/ .. \`@"
    /_| \__/ |_\
       \__U_/

```
the brace freeked me out at first
then I reslized what it reminded me of
now I brace all the time...}@{

**Comment by nextgenthemes** [comment]
I get the intention, but it's way overdone. Just a little bit more round and not THAT sharp.

**Comment by yvvt0379** [comment]
The curly braces is easy enough to tell apart from square & round brackets, which is good. However, they are too distractive, as if they are screaming “Hey look at me! I'm curly braces, not square brackets or round brackets!”

**Comment by jnfingerle** [comment]
I still don't get why people who don't like this font cannot just go on and use another one. This discussion clearly shows that there are a number of people who like the curlies just as they are. If this font isn't for you, fine, don't use it.

**Comment by yvvt0379** [comment]
But it's unwise to give up a font just because one disadvantage

**Comment by tkircher** [comment]
> I still don't get why people who don't like this font cannot just go on and use another one. This discussion clearly shows that there are a number of people who like the curlies just as they are. If this font isn't for you, fine, don't use it.

The purpose of the project, according to its own statement, is to provide a "font family that’s built with clarity, legibility, and the needs of developers in mind." In fact this one specific pair of glyphs ruins all of that for many of us. It is hard to look at, distracting, and comically unprofessional. That really isn't just an opinion. And suggesting that we should just get off your lawn doesn't acknowledge that a company as large as Intel has responsibilities to the community.

**Comment by yvvt0379** [comment]
Since opinions are so different, let OpenType features solve this problem. Thus, both sides could be satisfied.

**Comment by trinhnhatnam** [comment]
Please don't change it. Fonts with "normal looking" curly braces are abundant.

This font is for "low-vision developer audience". Please don't deviate from that mission. If folks don't like it, then they are probably not the target audience. They have plenty to choose from.

I don't have low-vision yet, but am loving this font. My previous favorite is Source Code Pro, but the latest versions don't have hinting, so they don't display well on Windows machines.
